packages/cli/src/commands/rewards/show.ts

```diff
@@ -0,0 +1,293 @@
+import { Validator, ValidatorGroup } from '@celo/contractkit/lib/wrappers/Validators'
+import { eqAddress } from '@celo/utils/lib/address'
+import { concurrentMap } from '@celo/utils/lib/async'
+import { flags } from '@oclif/command'
+import BigNumber from 'bignumber.js'
+import { cli } from 'cli-ux'
+import { EventLog } from 'web3/types'
+import { BaseCommand } from '../../base'
+import { newCheckBuilder } from '../../utils/checks'
+import { Flags } from '../../utils/command'
+
+interface VoterReward {
+  group: string
+  groupName: string
+  totalVoterPayment: BigNumber
+  blockNumber: number
+  activeAddressVotes?: BigNumber
+  activeGroupVotes?: BigNumber
+}
+
+interface ValidatorReward {
+  validator: string
+  validatorName: string
+  validatorPayment: BigNumber
+  validatorScore: BigNumber
+  group: string
+  blockNumber: number
+}
+
+interface ValidatorGroupReward {
+  group: string
+  groupName: string
+  validator: string
+  groupPayment: BigNumber
+  validatorScore: BigNumber
+  blockNumber: number
+}
+
+export default class Show extends BaseCommand {
+  static description =
+    'Show rewards information about a voter, registered Validator, or Validator Group'
+
+  static flags = {
+    ...BaseCommand.flags,
+    voter: Flags.address({ description: 'Voter to show rewards for' }),
+    validator: Flags.address({ description: 'Validator to show rewards for' }),
+    group: Flags.address({ description: 'Validator Group to show rewards for' }),
+    epochs: flags.integer({
+      default: 1,
+      description: 'Show results for the last N epochs',
+    }),
+  }
+
+  static args = []
+
+  static examples = ['show --address 0x5409ed021d9299bf6814279a6a1411a7e866a631']
+
+  async run() {
+    const res = this.parse(Show)
+    const election = await this.kit.contracts.getElection()
+    const validators = await this.kit.contracts.getValidators()
+    const epochSize = await validators.getEpochSize()
+    const currentBlock = await this.web3.eth.getBlockNumber()
+    const lastEpochBlock = Math.floor(currentBlock / epochSize) * epochSize
+    const fromBlock: number = lastEpochBlock - epochSize * ((res.flags.epochs || 1) - 1)
+    let voterRewards: VoterReward[] = []
+    let validatorRewards: ValidatorReward[] = []
+    let validatorGroupRewards: ValidatorGroupReward[] = []
+    let address = ''
+
+    if (res.flags.voter) {
+      address = res.flags.voter
+      await newCheckBuilder(this)
+        .isAccount(address)
+        .runChecks()
+    }
+
+    if (res.flags.validator) {
+      if (address) throw Error('Must select --voter, --validator, or --group')
+      address = res.flags.validator
+      await newCheckBuilder(this)
+        .isValidator(address)
+        .runChecks()
+    }
+
+    if (res.flags.group) {
```

Comment:
the code seems to imply that you could enter on more than one `if`, thus run more than one `runChecks()`. Or maybe they are exclusive, idk.

But if that's the case, i think it's better to runCheck() only once, thus, you can create the checksBuilder and use the `if`s to add checks conditionally, and then do only one `runChecks()`